test/functional/specs/desktop/C2584.js

```diff
@@ -0,0 +1,57 @@
+import { Selector } from "testcafe";
+
+const urlCollector = `http://127.0.0.1:8080/test/functional/sandbox/html/alloySdk.html`;
+
+fixture`C2584: Toggle logging through debug command`.page(urlCollector);
+
+test.meta({
+  ID: "C2584",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+test("Regression: debug command with enable: true. getLibraryInfo. refresh. toggle and repeat.", async t => {
+  await t
+    .click(Selector("#nologconfig-button"))
+    .click(Selector("#debugtrue-button"))
+    .click(Selector("#getlibraryinfo-button"));
+
+  let log = await t.getBrowserConsoleMessages();
+  await t.expect(log.log).match(/Executing getLibraryInfo command/);
+  let discard = log.log.length;
+
+  await t
+    .navigateTo(
+      "http://127.0.0.1:8080/test/functional/sandbox/html/alloySdk.html"
+    )
+    .click(Selector("#nologconfig-button"))
+    .click(Selector("#getlibraryinfo-button"));
+
+  log = await t.getBrowserConsoleMessages();
+  await t
+    .expect(log.log.slice(discard))
+    .match(/\[alloy] Executing getLibraryInfo command./);
+  discard = log.log.length;
```

Comment:
This seems repetitive and could probably be abstracted away. Maybe you could create a utility that test files can import with some sort of usage like this:

```
const consoleCursor = createConsoleCursor(t);

let newLogMessages = consoleCursor.getNewMessages().log;

... check new log messages
... do some user interactions

newLogMessages = consoleCursor.getNewMessages().log;

... check new log messages
... do some user interactions
```

Reply:
I added a logger utility, let me know what you think.